clap_derive/src/derives/subcommand.rs

```diff
@@ -180,16 +242,45 @@ fn gen_from_subcommand(
         }
     });
 
+    let wildcard = match ext_subcmd {
+        Some((span, var_name, str_ty, values_of)) => quote_spanned! { span=>
+            ("", ::std::option::Option::None) => ::std::option::Option::None,
+
+            (external, Some(matches)) => {
+                ::std::option::Option::Some(#name::#var_name(
+                    ::std::iter::once(#str_ty::from(external))
+                    .chain(
+                        matches.#values_of("").into_iter().flatten().map(#str_ty::from)
+                    )
+                    .collect::<::std::vec::Vec<_>>()
+                ))
+            }
+
+            (external, None) => {
+                ::std::option::Option::Some(#name::#var_name({
+                    let mut v = ::std::vec::Vec::with_capacity(1);
+                    v.push(#str_ty::from(external));
+                    v
+                }))
+            }
+        },
+
+        None => quote!(_ => None),
+    };
+
     quote! {
         fn from_subcommand<'b>(
             name: &'b str,
             sub: Option<&'b ::clap::ArgMatches>) -> Option<Self>
         {
             match (name, sub) {
-                #( #match_arms ),*,
+                #( #match_arms, )*
                 other => {
-                    #( #child_subcommands )*;
-                    None
+                    #( #child_subcommands )else*
```

Comment:
I don't understand the `else` here.

Reply:
This else serve as a separator for `if ` statements.

Before:
```rust
if let Some(...) = ... {
   //...
}
if let Some(...) = ... {
   //...
}
// ...
```

After:
```rust
if let Some(...) = ... {
   //...
} else if let Some(...) = ... {
   //...
}
// ...
```

The new variant looks better in cargo expand, this is the whole motivation